src/views/VideoView/VideoView.tsx

```diff
@@ -70,6 +78,24 @@ const VideoView: React.FC<RouteComponentProps> = () => {
     })
   }, [addVideoView, videoID])
 
+  // Save the video timestamp
+  useInterval(() => {
+    if (data?.video?.id && videoRef.current) {
+      updateWatchedVideos('INTERRUPTED', data.video.id, videoRef.current.currentTime)
+    }
+  }, 5000)
+
+  const handleVideoEnd = () => {
+    if (data?.video?.id) {
+      updateWatchedVideos('COMPLETED', data?.video?.id)
+    }
+  }
+  const handleVideoStart = () => {
+    if (videoTimestamp && videoRef.current) {
+      videoRef.current.currentTime = videoTimestamp
```

Comment:
This can be done by adding some prop like `startTime` or something to the video component